server/svix-server/src/v1/endpoints/application.rs

```diff
@@ -148,18 +151,26 @@ async fn get_application(
     Path(app_id): Path<ApplicationIdOrUid>,
     permissions: Permissions,
 ) -> Result<Json<ApplicationOut>> {
-    let app = application::Entity::secure_find_by_id_or_uid(permissions.org_id, app_id)
-        .one(db)
-        .await?
-        .ok_or_else(|| HttpError::not_found(None, None))?;
+    // If someone authorized for only applications calls this endpoint, it will only return their
+    // application regardless of input in the path
+    let app = application::Entity::secure_find_by_id_or_uid(
+        permissions.org_id,
+        permissions
+            .app_id
+            .map(|id| ApplicationIdOrUid(id.0))
+            .unwrap_or(app_id),
+    )
+    .one(db)
+    .await?
+    .ok_or_else(|| HttpError::not_found(None, None))?;
     Ok(Json(app.into()))
 }
 
 async fn update_application(
     Extension(ref db): Extension<DatabaseConnection>,
     Path(app_id): Path<ApplicationIdOrUid>,
     ValidatedJson(data): ValidatedJson<ApplicationIn>,
-    permissions: Permissions,
+    AuthenticatedOrganization { permissions }: AuthenticatedOrganization,
```

Comment:
It should be OrganizationAuthenticatedApp I think, and then you don't need to access permissions directly.

In general, accessing permissions directly should be less good now that we have the code that authenticates in the struct extension thing.